scripts/monitoring/auto_scale_workers.py

```diff
@@ -0,0 +1,256 @@
+        Returns the SQS Queue object
+    """
+    if settings.DEBUG or settings.TEST:
+        sqs = boto3.resource(
+            "sqs",
+            endpoint_url=os.environ.get("AWS_SQS_ENDPOINT", "http://sqs:9324"),
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        )
+    else:
+        sqs = boto3.resource(
+            "sqs",
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        )
+    if queue_name == "":
+        queue_name = "evalai_submission_queue"
+    # Check if the queue exists. If no, then create one
+    try:
+        queue = sqs.get_queue_by_name(QueueName=queue_name)
+    except botocore.exceptions.ClientError as ex:
+        if (
+            ex.response["Error"]["Code"]
+            != "AWS.SimpleQueueService.NonExistentQueue"
+        ):
+            raise Exception("Cannot get queue: {}".format(queue_name))
+    return queue
+
+
+def get_queue_name(param, challenge_pk):
+    """
+    Generate unique SQS queue name of max length 80 for a challenge
+
+    Arguments:
+        param {string} -- challenge title
+        challenge_pk {int} -- challenge primary key
+
+    Returns:
+        {string} -- unique queue name
+    """
+    # The max-length for queue-name is 80 in SQS
+    max_len = 80
+    max_challenge_title_len = 50
+
+    env = settings.ENVIRONMENT
+    queue_name = param.replace(" ", "-").lower()[:max_challenge_title_len]
+    queue_name = re.sub(r"\W+", "-", queue_name)
+
+    queue_name = "{}-{}-{}-{}".format(
+        queue_name, challenge_pk, env, uuid.uuid4()
+    )[:max_len]
+    return queue_name
+
+
+def get_queue_name_by_challenge(challenge):
+    challenge_id = challenge["id"]
+    challenge_title = challenge["title"]
+    return get_queue_name(challenge_id, challenge_title)
+
+
+def get_queue_length(queue):
+    # TODO: Check if we should use approximate number of messages
+    # https://github.com/boto/boto3/issues/599
+    # Ref: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/sqs.html#using-an-existing-queue
+    count = 0
+    for message in queue.receive_messages():
+        count += 1
+
+
+def get_queue_length_by_challenge(challenge):
+    queue_name = get_queue_name_by_challenge(challenge)
+    queue = get_sqs_queue_by_name(queue_name)
+    return get_queue_length(queue)
+
+
+def get_aws_credentials_for_challenge(challenge):
+    """
+    Return the AWS credentials for a challenge using challenge pk
+    Arguments:
+        challenge {dict} -- challenge json for which credentials are to be fetched
+    Returns:
+        aws_key {dict} -- Dict containing aws keys for a challenge
+    """
+    if challenge["use_host_credentials"]:
```

Comment:
we don't need to handle this case. For this iteration you can only do auto scaling for workers that are running at our end i.e.e ` challenge["use_host_credentials"]` is either none or false